pkg/api/handlers/decoder.go

```diff
@@ -143,3 +158,14 @@ func convertSignal(query string) reflect.Value {
 	}
 	return reflect.ValueOf(signal)
 }
+
+func convertOptionalBool(s string) reflect.Value {
+	if len(s) == 0 {
+		return reflect.ValueOf(types.OptionalBoolUndefined)
+	}
```

Comment:
Interesting this is not actually called when param missing, but that's ok since "zero" value is ok for us.

Reply:
What is the value of `s` here when param is missing ? is it `0` ?

Reply:
That's the thing: when the param is missing this function is not called at all. So the variable is untouched left in default state (types.OptionalBoolUndefined).

Reply:
I thought it would be called with an empty string so I added this if, but it appears it's not necessary.

Reply:
That is different to my original approach of implementing `UnmarshalText()`, the `UnmarshalText` function is called with an empty byte slice if the param is absent.

Reply:
well the `if` is not entirely pointless, it might apply if somebody deliberately sets the param to an empty string

Reply:
If variable is left with `types.OptionalBoolUndefined` when `param` is not defined then that is the correct behavior so I guess you can remove or keep this check as-is.

Reply:
Question is: what if caller deliberately uses empty string as query param, e.g.:
`http://v1.41/build?inheritlabels=` Is it interpreted as Undefined or False? It's fringe case so we don't have to care that much.

Reply:
>If variable is left with

The variable is untouched so whatever was assigned to it before calling `Decode()` will be there. We are not assigning it so it is `types.OptionalBoolUndefined` -- exactly what we want.

Reply:
Latter (not it this PR) we could consider making `SkipUnusedStages`, `CompatVolumes` or `IdentityLabel` also `types.OptionalBool` instead of plain `bool`.

Reply:
Latter (not it this PR) we could consider making `SkipUnusedStages`, `CompatVolumes` or `IdentityLabel` also `types.OptionalBool` instead of plain `bool`.

Reply:
@flouthoc thinking of it now probably all I had to do to fix this was:
adding 
`		InheritLabels:    true,` [here](https://github.com/containers/podman/blob/0c4d023822c4952c31c132c2479620f6c36aeaab/pkg/api/handlers/compat/images_build.go#L184)

Just single line fix!

But his is better since the underlying engine chooses interpretation of absent value.